client/components/inline-support-link/index.jsx

```diff
@@ -123,23 +125,30 @@ class InlineSupportLink extends Component {
 	}
 }
 
-const mapStateToProps = ( state ) => {
+const getLinkData = ( ownProps ) => {
+	const { supportContext } = ownProps;
+	const contextLinks = getContextLinks();
+	const linkData = contextLinks[ supportContext ];
+	if ( ! linkData ) {
+		return {};
+	}
+	return {
+		supportPostId: linkData.post_id,
+		supportLink: linkData.link,
+	};
+};
+
+const mapStateToProps = ( state, ownProps ) => {
 	return {
 		localeSlug: getCurrentLocaleSlug( state ),
+		...getLinkData( ownProps ),
```

Comment:
(Tracing this out loud to make sure I understand how it works)

If this component was called with a `supportContext` prop, and if there is a link and post ID corresponding to that context, then `getLinkData(ownProps)` here will overwrite any previously supplied values for `supportPostId` and `supportLink` due to how `connect` combines props.

Reply:
That's right. FWIW I believe this mechanism was designed by @pottedmeat (correct me if I'm wrong). If `supportContext` is provided, and there's a valid matching entry in `context-links.js`, any `supportPostId` or `supportLink` will be overridden. This logic could be inside the component itself, but putting it here has the (questionable?) advantage of leaving the component ignorant of this change.

It'd be kind of nice if there was a warning if someone accidentally tried to use both, but that might be out of the scope of this PR, especially when the intended change here is (as I understand it) to deprecate using `supportPostId` and `supportLink` directly.